packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart

```diff
@@ -116,6 +116,12 @@ class BitmapDescriptor {
     return BitmapDescriptor._(<dynamic>['fromBytes', byteData]);
   }
 
+  /// The inverse of .toJson.
+  /// Do not use this method, it can be removed at any time.
+  /// This is needed in Web to re-hydrate BitmapDescriptors that have been
+  /// transformed to JSON for transport.
+  BitmapDescriptor.fromJson(dynamic json) : _json = json;
```

Comment:
Can we add a TODO to link to an issue for removing this method?
Also, does it make sense to mark this as deprecated if we know we are going to remove it?

Reply:
Yes, I'll add a TODO to migrate the API of the google_maps_flutter plugin and link it here. It does make sense to be @Deprecated('Will be removed with: LINK_TO_THE_ISSUE').

I'll change this right away.

Reply:
cc @amirh 
Is it ok to add deprecation notes on a temporary API that we just add.

Reply:
This is the crux of the issue: https://github.com/flutter/flutter/issues/70330

Reply:
We discussed this offline. I'll add some validation to the fromJson method to try to ensure that users can't create objects substantially different than those by the static methods, and remove the deprecation notice.

The bigger problem here is that the API is currently promising to return a JSON object with certain format, and not really the way it's created.